pkg/storage/unified/sql/data/resource_history_list.sql

```diff
@@ -23,10 +23,29 @@ SELECT
             {{ end }}
         GROUP BY mkv.{{ .Ident "namespace" }}, mkv.{{ .Ident "group" }}, mkv.{{ .Ident "resource" }}, mkv.{{ .Ident "name" }} 
     ) AS maxkv
-    ON maxkv.{{ .Ident "guid" }}  = kv.{{ .Ident "guid" }} 
+    ON
+        maxkv.{{ .Ident "resource_version" }}  = kv.{{ .Ident "resource_version" }}
+        AND maxkv.{{ .Ident "namespace" }}     = kv.{{ .Ident "namespace" }}
+        AND maxkv.{{ .Ident "group" }}         = kv.{{ .Ident "group" }}
+        AND maxkv.{{ .Ident "resource" }}      = kv.{{ .Ident "resource" }}
+        AND maxkv.{{ .Ident "name" }}          = kv.{{ .Ident "name" }}
     WHERE kv.{{ .Ident "action" }}  != 3 
-    ORDER BY kv.{{ .Ident "resource_version" }} ASC
+    {{ if and .Request.Options .Request.Options.Key }}
+        {{ if .Request.Options.Key.Namespace }}
+        AND kv.{{ .Ident "namespace" }} = {{ .Arg .Request.Options.Key.Namespace }}
+        {{ end }}
+        {{ if .Request.Options.Key.Group }}
+        AND kv.{{ .Ident "group" }}     = {{ .Arg .Request.Options.Key.Group }}
+        {{ end }}
+        {{ if .Request.Options.Key.Resource }}
+        AND kv.{{ .Ident "resource" }}  = {{ .Arg .Request.Options.Key.Resource }}
+        {{ end }}
+        {{ if .Request.Options.Key.Name }}
+        AND kv.{{ .Ident "name" }}      = {{ .Arg .Request.Options.Key.Name }}
+        {{ end }}
+    {{ end }}
+    ORDER BY kv.{{ .Ident "resource_version" }} DESC
     {{ if (gt .Request.Limit 0) }}
-    LIMIT {{ .Arg .Request.Offset }}, {{ .Arg .Request.Limit }}
+    LIMIT {{ .Arg .Request.Limit }} OFFSET {{ .Arg .Request.Offset }}
```

Comment:
Aren't both limit-offset syntax here correct? Was there an issue with certain dialects?

Reply:
`LIMIT` is non-standard. The SQL standard is `OFFSET m FETCH FIRST n ROWS ONLY`, but is only supported by PostgreSQL out of our 3 supported dialects. The `LIMIT m, n` is a MySQL idiom that SQLite adopted for compatibility with MySQL. Conversely, the `LIMIT n OFFSET m` was a PostgreSQL idiom which was later ported to MySQL and transitively to SQLite for compatibility, so it's our best choice here.